lib/src/screens/debug/scheme_management/scheme_manager_detail_screen.dart

```diff
@@ -0,0 +1,253 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_i18n/flutter_i18n.dart';
+import 'package:rxdart/rxdart.dart';
+
+import '../../../models/authentication_events.dart';
+import '../../../models/enrollment_events.dart';
+import '../../../models/irma_configuration.dart';
+import '../../../models/scheme_events.dart';
+import '../../../theme/theme.dart';
+import '../../../widgets/active_indicator.dart';
+import '../../../widgets/irma_app_bar.dart';
+import '../../../widgets/irma_bottom_bar.dart';
+import '../../../widgets/irma_icon_button.dart';
+import '../../../widgets/irma_repository_provider.dart';
+import '../../enrollment/provide_email/provide_email_screen.dart';
+import '../../error/error_screen.dart';
+import '../../pin/yivi_pin_screen.dart';
+import '../util/snackbar.dart';
+
+class SchemeManagerDetailScreen extends StatelessWidget {
+  final SchemeManager schemeManager;
+  final bool isActive;
+
+  final String? appId;
+
+  const SchemeManagerDetailScreen(
+    this.schemeManager,
+    this.isActive,
+    this.appId,
+  );
+
+  Future<String?> _requestPin(BuildContext context, String title, String instruction) async {
+    final repo = IrmaRepositoryProvider.of(context);
+    final navigator = Navigator.of(context);
+
+    final hasLongPin = await repo.preferences.getLongPin().first;
+    final maxPinSize = hasLongPin ? 16 : 5;
+
+    return navigator.push(
+      MaterialPageRoute(
+        builder: (context) => YiviPinScaffold(
+          appBar: IrmaAppBar(title: title, hasBorder: false),
+          body: YiviPinScreen(
+            instruction: instruction,
+            pinBloc: EnterPinStateBloc(maxPinSize),
+            maxPinSize: maxPinSize,
+            onSubmit: (pin) => navigator.pop(pin),
+            listener: (context, state) {
+              if (!hasLongPin && state.pin.length == 5) {
+                navigator.pop(state.toString());
+              }
+            },
+          ),
+        ),
+      ),
+    );
+  }
+
+  Future<void> _verifyPin(BuildContext context, String schemeId) async {
+    final repo = IrmaRepositoryProvider.of(context);
+    final navigator = Navigator.of(context);
+
+    final pin = await _requestPin(
+      context,
+      FlutterI18n.translate(context, 'debug.scheme_management.verify_pin.title'),
+      FlutterI18n.translate(
+        context,
+        'debug.scheme_management.verify_pin.content',
+        translationParams: {
+          'scheme': schemeManager.id,
+        },
+      ),
+    );
+    if (pin == null) return;
+
+    repo.bridgedDispatch(AuthenticateEvent(pin: pin, schemeId: schemeId));
+
+    final event = await repo.getEvents().whereType<AuthenticationEvent>().first;
+    navigator.pop();
```

Comment:
In `_requestPin` we also do a `navigator.pop()`. This means that we pop back to the `SchemeManagementScreen` now after the PIN was verified. Was this on purpose?